daml/splice-amulet/daml/Splice/AmuletRules.daml

```diff
@@ -1597,6 +1612,42 @@ splitAndBurn sender amount inputs context dso usage = do
 totalBurnFromSummary : TransferSummary -> Decimal
 totalBurnFromSummary TransferSummary{..} = sum outputFees + holdingFees + senderChangeFee
 
+payForTransferPreapproval
+  : Party -> Party -> PaymentTransferContext -> [TransferInput] -> Time -> Time
+  -> Update (Decimal, TransferResult, Metadata)
+payForTransferPreapproval dso provider context inputs expiresAt newExpiresAt = do
+  amuletRules <- fetchChecked (ForDso dso) context.amuletRules
+  configUsd <- getValueAsOfLedgerTime amuletRules.configSchedule
+  let extension = newExpiresAt `subTime` expiresAt
+  let paidExtension = extension - getTransferPreapprovalBaseDuration configUsd
+  if paidExtension <= aunit
```

Comment:
@moritzkiefer-da : can you keep me honest here: it seems to me that this needs to work slightly differently to be backwards compatible: if more than one input is specified, then a transfer must happen so that the inputs are merged, correct?

Reply:
OK. Let me then adjust the `Wallet.Install` contract, which currently assumes it.